### ISSUE TITLE
Xcode 8.3 issue with class_getSuperClass

### ISSUE THREAD

**Issue by simonstewart** [net-a-porter-mobile/XCTest-Gherkin#56]
All's been fine until this morning's update to Xcode 8.3.

Now the UI tests are failing with a "exc_bad_access" at line 35 of ClassHelperMethods.swift:
"if class_getSuperclass(currentClass) is T"

Enabling Zombie objects didn't give any additional output.

Anyone else using Xcode 8.3?

**Comment by nap-sam-dean** [comment]
Hey.

It's very strange, this method seems to have stopped working for WebKit classes :| Anyway, I've fixed it in this PR - https://github.com/net-a-porter-mobile/XCTest-Gherkin/pull/57

Can you take a look and see if this is OK?

Thanks,
S

**Comment by simonstewart** [comment]
@nap-sam-dean glad I'm not the only one!

**Comment by nap-sam-dean** [comment]
I've also made a SO question to see if anyone else has seen this issue: https://stackoverflow.com/questions/43073087/crash-casting-wknsurlrequest-as-other-type

**Comment by kerrmarin** [comment]
🤔  just a thought here, but looking at that question it looks like `WKNSURLRequest` is a "private" class. If you do the same for any other WebKit class that's actually exported by the WebKit module there is no crash. e.g.:

```
import Foundation
import WebKit

final class Sigh: NSObject { }

NSClassFromString("NSObject") as? Sigh.Type

NSClassFromString("WKBackForwardList") as? Sigh.Type
NSClassFromString("WKBackForwardListItem") as? Sigh.Type
NSClassFromString("WKFoundation") as? Sigh.Type
NSClassFromString("WKError") as? Sigh.Type
NSClassFromString("WKFrameInfo") as? Sigh.Type
NSClassFromString("WKNavigation") as? Sigh.Type
NSClassFromString("WKNavigationAction") as? Sigh.Type
NSClassFromString("WKNavigationDelegate") as? Sigh.Type
NSClassFromString("WKNavigationResponse") as? Sigh.Type
NSClassFromString("WKOpenPanelParameters") as? Sigh.Type
NSClassFromString("WKPreferences") as? Sigh.Type
NSClassFromString("WKPreviewActionItem") as? Sigh.Type
NSClassFromString("WKPreviewActionItemIdentifiers") as? Sigh.Type
NSClassFromString("WKPreviewElementInfo") as? Sigh.Type
```

**Comment by nap-sam-dean** [comment]
I thought that, but why only WebKit private classes - I'm listing through _all_ the loaded classes and I just don't believe they're all public :) And I couldn't think of a way to exclude private classes before I tried to cast them :| There's something special going on inside these classes that isn't in any others :|

**Comment by nap-sam-dean** [comment]
I've released version 0.10.0 with this fix in - @simonstewart can you update the pods in your app, and if this is fixed please close this issue? Thanks!

**Comment by nap-sam-dean** [comment]
We can't reproduce this after merging #69, so marking this as closed.
